### ISSUE TITLE
Unable to use maven site plugin when having spring-boot-starter-parent as the parent pom

### ISSUE THREAD

**Issue by trung** [spring-projects/spring-boot#3358]
In my pom.xml, i declare

    <parent>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-parent</artifactId>
       <version>1.2.1.RELEASE</version>
    </parent>

when I run `mvn site:run`, I receive

    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD FAILURE
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time: 16.089 s
    [INFO] Finished at: 2015-06-30T13:21:33+08:00
    [INFO] Final Memory: 30M/437M
    [INFO] ------------------------------------------------------------------------
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.4:run (default-cli) on project otm-solr-main: Unable to set up webapp: SiteToolException: The site descriptor cannot be resolved from the repository: ArtifactResolutionException: Unable to locate site descriptor: Could not transfer artifact org.springframework.boot:spring-boot-starter-parent:xml:site_en:1.2.1.RELEASE ..<snip>..

I've tried the following and it works:
 * Put a default `site_en.xml` in `spring-boot-starter-parent/src/site` and `spring-boot-dependencies/src/site` 
 * Add the `maven-site-plugin` `attach-descriptor` in pom.xml for the above projects as per http://maven.apache.org/plugins/maven-site-plugin/maven-3.html#Site_descriptor_attachment
 * Rerun `mvn install` to have it deployed to local repo

Shall i create a pull request?

**Comment by philwebb** [comment]
Adding `site_en.xml` to `spring-boot-starter-parent` feels like the wrong way to fix this. We only declare the `maven-site-plugin` in the `pluginManagement` section. What happens if you add a `<relativePath/>` element to your parent import and declare the `maven-site-plugin` directly?

```xml
...
<project>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>1.3.0.BUILD-SNAPSHOT</version>
		<relativePath/>
	</parent>
	...
	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-site-plugin</artifactId>
			</plugin>
			...
		</plugins>
	</build>
</project>
```

@snicoll any ideas why the site plugin would need this file?

**Comment by snicoll** [comment]
I have tried on a sample project from start.spring.io and `mvn site` ran as expected. You probably have something special in your project that makes it fail but I hardly find how Spring Boot could be responsible of that, really. 

Please raise this on stackoverflow with more details about your setup. Thanks.

**Comment by trung** [comment]
Hmm, it only happened in my company environment where dependencies can only be obtained from official internal repo. Let me try again and get back.

**Comment by ogg1e** [comment]
@trung I'm seeing this same issue with an internal repo as well. It's specific to spring boot since all our other projects successfully run with site:site. I'm not saying spring boot is at fault here, but there's something special about it that causes this error.

**Comment by pgpx** [comment]
@trung @ogg1e I get the same problem - using an ancient version of Archiva (1.3.5) as our caching repo, the site_en.xml file isn't found, but it works ok if I disable that and access Maven Central directly. I suspect that my repository just doesn't understand site descriptors (and that it isn't a Spring Boot issue).  This was easy to test using a project created on http://start.spring.io that @snicoll tested.

**Comment by ogg1e** [comment]
@pgpx I'm also using archiva, but I'm on version 2.2.0
